### PR TITLE
Fix a regression in vite.build() base option

### PR DESCRIPTION
**Pull request by FredKSchott**
## Changes

- Fixes this regression: https://github.com/withastro/astro/pull/2486/files#r795260099
- no changeset needed since this regression hasn't gone out yet

## Testing

- Does this need a test? Odd that none of our build tests caught this, since it feels like any buildOptions.site value would break this.

## Docs

- N/A

### CODE REVIEW

**Comment by matthewp** [comment]
> Does this need a test? Odd that none of our build tests caught this, since it feels like any buildOptions.site value would break this.

This is a static build bug and most of the tests aren't using that. There are a few though, so I'll update one of them to use it.

**Comment by matthewp** [comment]
Added usage of subpath to one of the static build tests, fixes some other places where it wasn't working. Should be good now!

**Comment by matthewp** [comment]
Looking into the Windows test breaking...

**Comment by matthewp** [comment]
Ok, all appears to be good.

File: packages/astro/src/runtime/server/index.ts [hunk 1]
```diff
@@ -263,6 +263,10 @@ If you're still stuck, please open an issue on GitHub or join us at https://astr
 
 /** Create the Astro.fetchContent() runtime function. */
 function createFetchContentFn(url: URL, site: URL) {
+	let sitePathname = site.pathname;
+	if(!sitePathname.endsWith('/')) {
```
Comment (FredKSchott) [inline packages/astro/src/runtime/server/index.ts:267]: I made a change in #2471 so that this is always true (`site` always ends with `/`). Check is no longer needed!

**Review by FredKSchott — COMMENTED** [review]
@matthewp LGTM with 1 small comment, feel free to merge when ready!

File: packages/astro/src/runtime/server/index.ts (hunk 1 above)
Comment (matthewp) [reply packages/astro/src/runtime/server/index.ts:267]: I'll test again, but this was indeed causing an issue.

Comment (FredKSchott) [reply packages/astro/src/runtime/server/index.ts:267]: For reference, this is the line that should make that moot: https://github.com/withastro/astro/blob/c9bb1147cbfae20e3ecdf29ef2866a183b3b18e3/packages/astro/src/core/config.ts#L59

Comment (matthewp) [reply packages/astro/src/runtime/server/index.ts:267]: It's because your tests add their own config that doesn't go through this schema stuff. I'll see if I can fix that.

Comment (FredKSchott) [reply packages/astro/src/runtime/server/index.ts]: Ah, I wonder if we can make the tests go through the "user config -> final config" pipeline, instead of just submitting their own final config. A PR for another time, I guess

Comment (matthewp) [reply packages/astro/src/runtime/server/index.ts]: Yes, I thought about doing this, but worried it might break a bunch of tests. I might try in another branch just to see.
